domain-specific-datasets/app/app.py

```diff
@@ -0,0 +1,124 @@
+import json
+
+from pytest import mark
+import streamlit as st
+
+from hub import push_dataset_to_hub
+from infer import query
+from defaults import (
+    DEFAULT_DOMAIN,
+    DEFAULT_PERSPECTIVES,
+    DEFAULT_TOPICS,
+    DEFAULT_EXAMPLES,
+    DEFAULT_SYSTEM_PROMPT,
+    N_PERSPECTIVES,
+    N_TOPICS,
+    N_EXAMPLES,
+    SEED_DATA_PATH,
+    PIPELINE_PATH,
+)
+from pipeline import serialize_pipeline
+
+st.markdown("# 🧑‍🌾 Domain Data Grower")
+st.markdown("## 🌱 Create a dataset seed for aligning models to a specific domain")
+st.markdown(
+    "This app helps you create a dataset seed for building diverse domain-specific datasets for aligning models."
+)
+st.markdown(
+    "Alignment datasets are used to fine-tune models to a specific domain or task, but as yet, there's a shortage of diverse datasets for this purpose."
+)
+
+project_name = st.text_input("Project Name", DEFAULT_DOMAIN)
+domain = st.text_input("Domain", DEFAULT_DOMAIN)
+hub_username = st.text_input("Hub Username", "argilla")
+hub_token = st.text_input("Hub Token", type="password")
+argilla_url = st.text_input("Argilla API URL", "https://argilla-farming.hf.space")
+argilla_api_key = st.text_input("Argilla API Key", "owner.apikey")
+
+st.header("👩🏼‍🔬 Domain Expert")
+st.header("Define the domain expertise that you want to train a language model on.")
+domain_expert_prompt = st.text_area(
+    label="Domain Expertise",
+    value=DEFAULT_SYSTEM_PROMPT,
+)
+st.header("👯️ Domain Perspectives")
+st.markdown(
+    "Perspectives are different viewpoints or angles from which a domain can be viewed. For example, the domain of farming can be viewed from the perspective of a commercial farmer or an independent family farmer."
+)
+
+perspectives = [
+    st.text_input(f"Domain Perspective {n}", value=DEFAULT_PERSPECTIVES[n])
+    for n in range(N_PERSPECTIVES)
+]
+
+st.header("🧺️ Domain Topics")
+st.markdown(
+    "Topics are the main themes or subjects that are relevant to the domain. For example, the domain of farming can have topics like soil health, crop rotation, or livestock management."
+)
+topics = [
+    st.text_input(f"Domain Topic {n}", value=DEFAULT_TOPICS[n]) for n in range(N_TOPICS)
+]
+
+st.header("📚 Examples")
+
+questions_answers = []
+
```

Comment:
should we consider adding calling the API with an example the data in order to have an idea and allow for prompt engineering within the streamlit while they are fine-tuning their input?